pandas/tests/indexes/multi/test_copy.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+from copy import copy, deepcopy
+
+import pandas.util.testing as tm
+from pandas import (CategoricalIndex, IntervalIndex, MultiIndex, PeriodIndex,
+                    RangeIndex, Series, compat)
+
+
+def assert_multiindex_copied(copy, original):
+    # Levels should be (at least, shallow copied)
+    tm.assert_copy(copy.levels, original.levels)
+    tm.assert_almost_equal(copy.labels, original.labels)
+
+    # Labels doesn't matter which way copied
+    tm.assert_almost_equal(copy.labels, original.labels)
+    assert copy.labels is not original.labels
+
+    # Names doesn't matter which way copied
+    assert copy.names == original.names
+    assert copy.names is not original.names
+
+    # Sort order should be copied
+    assert copy.sortorder == original.sortorder
+
+
+def test_copy(idx):
+    i_copy = idx.copy()
+
+    assert_multiindex_copied(i_copy, idx)
+
+
+def test_shallow_copy(idx):
+    i_copy = idx._shallow_copy()
+
+    assert_multiindex_copied(i_copy, idx)
+
+
+def test_view(idx):
+    i_view = idx.view()
+    assert_multiindex_copied(i_view, idx)
+
+
+def test_copy_name(idx):
+    # gh-12309: Check that the "name" argument
+    # passed at initialization is honored.
+
+    # TODO: Remove or refactor MultiIndex not tested.
```

Comment:
yep refactor this

Reply:
I wrote a new test specifically for MultiIndexes. I test the levels, names, and the labels are honored when copied. I'm not sure about the dtype though. Looks like this is always an object anyway.